### DISCUSSION TITLE
Queue Worker Doesn't Work In Two Containers

### DISCUSSION THREAD

**Discussion by kresnasatya** [serversideup/docker-php#225] [Q&A] [answered]
Hi everyone, currently I try to implement Laravel Job Batching using Laravel Queue Worker in Docker. Based on [ServerSide Up's documentation](https://serversideup.net/open-source/docker-php/docs/laravel/laravel-queue), it recommends to run queue worker on separate container. So, I have two containers: one for laravel web app and another one for laravel queue worker. Those containers use the same docker image. I create Docker image that wrap serversideup/docker-php (php8.2-fpm-nginx) with my Laravel web app.

When I try to test job batching with those containers, the queue worker doesn't work as video below.

![Two Containers](https://github.com/serversideup/docker-php/assets/11099186/2b4a0e4b-f075-42e6-9530-684ed0fda188)

But, when I just use one container the queue worker works as video below.

![One Containers](https://github.com/serversideup/docker-php/assets/11099186/95e7d40e-b251-4a9f-88c4-2d199d39ed0e)

Here's my repository that reproduce those steps above: https://github.com/kresnasatya/livewire-usecase

I'm using Laravel version 10.

I'm curious what are the things that I'm missing in order to make queue worker works on separate containers.

**Comment by minkbear** [comment]
what is QUEUE_CONNECTION you using?

**Reply by kresnasatya** [reply]
I'm using database as QUEUE_CONNECTION.

**Reply by minkbear** [reply]
I think your issue is not about "serversideup/docker-php".

because in https://github.com/kresnasatya/livewire-usecase/blob/main/docker-compose.yml queue:work is not specify connection in https://github.com/kresnasatya/livewire-usecase/blob/main/config/queue.php

you have to specify connection for example 
`php artisan queue:work database --tries=3`

Please take a look at https://laravel.com/docs/10.x/queues#specifying-the-connection-queue

**Reply by kresnasatya** [reply]
Hi, unfortunately I try it but it doesn't work as video below.

![Screen Recording 2023-11-30 at 20 17 10](https://github.com/serversideup/docker-php/assets/11099186/f233ab14-790b-4542-9b6f-0555112b29e6)

**Reply by minkbear** [reply]
Check you database table "jobs" to make sure dispatch is work.

**Reply by kresnasatya** [reply, marked as answer]
The jobs records saved in app container and doesn't share record with queue container. Hmm, I think it's because of the database I'm using is sqlite instead of MySQL or Redis. I will try it later. Here's the gif.

Update: I'm using MySQL as DATABASE_CONNECTION and Redis as QUEUE_CONNECTION. Now, it works! I don't use SQLite because the database doesn't share connection or data between two containers.

![Screen Recording 2023-12-01 at 07 20 53](https://github.com/serversideup/docker-php/assets/11099186/92fc7304-83b4-4c50-9f09-24bd8d11168a)

**Reply by minkbear** [reply]
Congrats, you got it. 
please take a look on the network of docker. 
if you change queue connection to redis is not solved because app, queue, redis container is not the same network. 

please take a look https://www.tutorialworks.com/container-networking/
hope this help.

**Reply by kresnasatya** [reply]
Thanks a lot @minkbear.
